x/protocolpool/keeper/keeper.go

```diff
@@ -90,6 +102,126 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
 
+func (k Keeper) withdrawContinuousFund(ctx context.Context, recipient sdk.AccAddress) (sdk.Coin, error) {
+	denom, err := k.iterateAndUpdateFundsDistribution(ctx)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while iterating all the continuous funds: %w", err)
+	}
+	// get allocated continuous fund
+	fundsAllocated, err := k.getAndResetDistributedFunds(ctx, recipient)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while getting distributed funds: %w", err)
+	}
+
+	cf, err := k.ContinuousFund.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no continuous fund found for recipient: %s", recipient.String())
+		}
+		return sdk.Coin{}, err
+	}
+
+	recipientBal := k.bankKeeper.GetAllBalances(ctx, recipient)
+	recipientAmount := recipientBal.AmountOf(denom)
+	totalRecipientBal := recipientAmount.Add(fundsAllocated)
+	// check if the recipient account balance exceeds maxDistributedCapital after distribution
+	var withdrawnAmount sdk.Coin
+	if totalRecipientBal.LT(cf.MaxDistributedCapital) {
+		withdrawnAmount = sdk.NewCoin(denom, fundsAllocated)
+		// Distribute funds to the recipient from pool module account
+		err := k.DistributeFromCommunityPool(ctx, sdk.NewCoins(withdrawnAmount), recipient)
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+
+		// decrement fundsAllocated from ToDistribute
+		toDistribute, err := k.ToDistribute.Get(ctx)
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+		err = k.ToDistribute.Set(ctx, toDistribute.Sub(fundsAllocated))
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+	}
+
+	return withdrawnAmount, nil
+}
+
+func (k Keeper) iterateAndUpdateFundsDistribution(ctx context.Context) (string, error) {
+	totalPercentageToBeDistributed := math.ZeroInt()
+	err := k.RecipientFundPercentage.Walk(ctx, nil, func(key sdk.AccAddress, value math.Int) (stop bool, err error) {
+		totalPercentageToBeDistributed = totalPercentageToBeDistributed.Add(value)
+		return false, nil
+	})
+	if err != nil {
+		return "", err
+	}
+	if totalPercentageToBeDistributed.GT(math.NewInt(100)) {
+		return "", fmt.Errorf("total funds percentage cannot exceed 100")
+	}
+
+	poolMAcc := k.authKeeper.GetModuleAccount(ctx, types.ModuleName)
+
+	distrMAcc := k.authKeeper.GetModuleAccount(ctx, distrtypes.ModuleName)
+	distrBal := k.bankKeeper.GetAllBalances(ctx, distrMAcc.GetAddress())
```

Comment:
how come we are checking distribution? are the funds not in the protocol pool account?